src/VirtualClient/VirtualClient.Actions/DXMicrobenchmarks/DXFLOPSExecutor.cs

```diff
@@ -0,0 +1,225 @@
+
+            }
+        }
+
+        /// <summary>
+        /// Number of data elements
+        /// </summary>
+        protected int NumDataElements
+        {
+            get
+            {
+                return this.Parameters.GetValue<int>(nameof(this.NumDataElements), 1048576);
+
+            }
+        }
+
+        /// <summary>
+        /// Number of data elements
+        /// </summary>
+        protected string Precision
+        {
+            get
+            {
+                return this.Parameters.GetValue<string>(nameof(this.Precision), "f32").ToLower();
+            }
+        }
+
+        /// <summary>
+        /// Defines the path to the package that contains the workload
+        /// executable.
+        /// </summary>
+        protected DependencyPath Package { get; set; }
+
+        /// <summary>
+        /// Path to the workload executable
+        /// </summary>
+        protected string ExecutablePath { get; set; }
+
+        /// <summary>
+        /// Runs the workload
+        /// </summary>
+        protected override Task ExecuteAsync(EventContext telemetryContext, CancellationToken cancellationToken)
+        {
+            return this.ExecuteWorkloadAsync(telemetryContext, cancellationToken);
+        }
+
+        /// <summary>
+        /// Initializes the environment
+        /// </summary>
+#pragma warning disable AsyncFixer01 // Unnecessary async/await usage
+        protected override async Task InitializeAsync(EventContext telemetryContext, CancellationToken cancellationToken)
+        {
+            PlatformSpecifics.ThrowIfNotSupported(this.Platform);
+
+            await this.InitializePackageLocationAsync(cancellationToken)
+                .ConfigureAwait(false);
+
+            this.ExecutablePath = this.PlatformSpecifics.Combine(this.Package.Path, "Benchmarks", "GPUCore.exe");
+        }
+#pragma warning restore AsyncFixer01 // Unnecessary async/await usage
+
+        private IList<Metric> CaptureResults(IProcessProxy workloadProcess, string output, EventContext telemetryContext)
+        {
+            if (workloadProcess.ExitCode == 0)
+            {
+                DXFLOPSParser resultsParser = new DXFLOPSParser(output);
+                return resultsParser.Parse();
+            }
+            else
+            {
+                return new List<Metric>();
+            }
+
+        }
+
+        private Task ExecuteWorkloadAsync(EventContext telemetryContext, CancellationToken cancellationToken)
+        {
+            IList<Metric> metrics = new List<Metric>();
+
+            EventContext relatedContext = telemetryContext.Clone()
+                .AddContext("executable", this.ExecutablePath);
+
+            return this.Logger.LogMessageAsync($"{nameof(DXFLOPSParser)}.ExecuteWorkload", relatedContext, async () =>
+            {
+                using (BackgroundOperations profiling = BackgroundOperations.BeginProfiling(this, cancellationToken))

```

Comment:
You don't have to use this profiling. This was only used in some tools that the AEP team wants linux profiler running at the same time, I don't imagine you need it